src/read.c

```diff
@@ -1175,11 +1181,19 @@ void ReadFuncExpr (
 	    ASS_LIST( nams, narg+nloc, name );
 	    Match(S_IDENT,"identifier",S_RPAREN|S_LOCAL|STATBEGIN|S_END|follow);
 	}
+
+	if(TLS(Symbol) == S_DOTDOT) {
+	    SyntaxError("Three dots required for variadic argument list");
+	}
+	if(TLS(Symbol) == S_DOTDOTDOT) {
+	    isvarg = 1;
+	    GetSymbol();
+    }
+
 	while ( TLS(Symbol) == S_COMMA ) {
-	    if (narg > 0 && !strcmp(CSTR_STRING(ELM_LIST(nams,narg)),"arg"))
-	      {
-		SyntaxWarning("arg used not as the last argument");
-	      }
+	    if (isvarg) {
+	        SyntaxError("Only final argument can be variadic");
+	    }
```

Comment:
Well, don't waste too much time on it; changing this is certainly not a priority, just "would be nice to have". But there are many other things that would be nice to have, and many of them would be much, much nicer (such as what we get by merging this PR ... ;-).